CompuCell3D/core/CompuCell3D/plugins/FocalPointPlasticity/FocalPointPlasticityLinks.h

```diff
@@ -0,0 +1,300 @@
+		const int getNeighborOrder() { return fppltd.neighborOrder; }
+		// Set neighbor order
+		void setNeighborOrder(int _neighborOrder) { fppltd.neighborOrder = _neighborOrder; }
+		// Get is anchor
+		const bool isAnchor() { return fppltd.anchor; }
+		// Get initialization step
+		const int getInitMCS() { return fppltd.initMCS; }
+		// Get first id
+		virtual const long getId0() { return long(0); };
+		// Get second id
+		virtual const long getId1() { return long(0); };
+		// Get first object
+		CellG* getObj0() { return initiator; }
+		// Get second object
+		CellG* getObj1() { return initiated; }
+	};
+
+	class FOCALPOINTPLASTICITY_EXPORT FocalPointPlasticityLink : public FocalPointPlasticityLinkBase {
+
+		FocalPointPlasticityLinkType type = FocalPointPlasticityLinkType::REGULAR;
+
+	public:
+		FocalPointPlasticityLink() {};
+		FocalPointPlasticityLink(CellG *_initiator, CellG *_initiated, Potts3D *_potts, FocalPointPlasticityLinkTrackerData _fppltd)
+		{
+			initiator = _initiator;
+			initiated = _initiated;
+			potts = _potts;
+			fppltd = FocalPointPlasticityLinkTrackerData(_fppltd);
+			fppltd.anchor = false;
+
+			DerivedProperty<FocalPointPlasticityLinkBase, float, &FocalPointPlasticityLinkBase::getDistance> length(this);
+			DerivedProperty<FocalPointPlasticityLinkBase, float, &FocalPointPlasticityLinkBase::getTension> tension(this);
+		}
+		FocalPointPlasticityLink(CellG *_initiator, CellG *_initiated, Potts3D *_potts, FocalPointPlasticityTrackerData _fpptd) :
+			FocalPointPlasticityLink(_initiator, _initiated, _potts, FocalPointPlasticityLinkTrackerData(_fpptd)) {}
+		FocalPointPlasticityLink(CellG *_initiator, CellG *_initiated, Potts3D *_potts, float _lambdaDistance = 0.0, float _targetDistance = 0.0, float _maxDistance = 100000.0, int _initMCS = 0) :
+			FocalPointPlasticityLink(_initiator, _initiated, _potts, FocalPointPlasticityLinkTrackerData(_lambdaDistance, _targetDistance, _maxDistance, _initMCS))
+		{}
+
+		const long getId0() { return initiator->id; }
+		const long getId1() { return initiated->id; }
+
+	};
+
+	class FOCALPOINTPLASTICITY_EXPORT FocalPointPlasticityInternalLink : public FocalPointPlasticityLinkBase {
+
+		FocalPointPlasticityLinkType type = FocalPointPlasticityLinkType::INTERNAL;
+
+	public:
+		FocalPointPlasticityInternalLink() {};
+		FocalPointPlasticityInternalLink(CellG *_initiator, CellG *_initiated, Potts3D *_potts, FocalPointPlasticityLinkTrackerData _fppltd)
+		{
+			initiator = _initiator;
+			initiated = _initiated;
+			potts = _potts;
+			fppltd = FocalPointPlasticityLinkTrackerData(_fppltd);
+			fppltd.anchor = false;
+
+			DerivedProperty<FocalPointPlasticityLinkBase, float, &FocalPointPlasticityLinkBase::getDistance> length(this);
+			DerivedProperty<FocalPointPlasticityLinkBase, float, &FocalPointPlasticityLinkBase::getTension> tension(this);
+		}
+		FocalPointPlasticityInternalLink(const CellG *_initiator, CellG *_initiated, Potts3D *_potts, FocalPointPlasticityLinkTrackerData _fppltd) :
```

Comment:
I would avoid overloads on constantness

Reply:
There is way to many construtors here

Reply:
Implemented in https://github.com/CompuCell3D/CompuCell3D/pull/361/commits/1305c536c83539572ee8f043066638c4d3c5e949